pkg/openstack/glance.go

```diff
@@ -46,20 +45,16 @@ func ReconcileGlance(ctx context.Context, instance *corev1beta1.OpenStackControl
 
 	// add selector to service overrides
 	for name, glanceAPI := range instance.Spec.Glance.Template.GlanceAPIs {
-		var currentAPI glancev1.GlanceAPITemplate
 		for _, endpointType := range []service.Endpoint{service.EndpointPublic, service.EndpointInternal} {
-			currentAPI = instance.Spec.Glance.Template.GlanceAPIs[name]
```

Comment:
@stuggi I suspect the panic:

```
"namespace": "openstack-kuttl-tests", "name": "openstack-basic", "reconcileID": "aa0a7c5f-d84c-4d26-b919-0dde72f4a13e"}
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

comes from this part where you iterate over the `glanceAPI` instance and use this object directly instead of `currentAPI`.
The advantage of that variable (that shouldn't hurt in the whole logic) is to have a way to properly reference the current entry in the array (`currentAPI = instance.Spec.Glance.Template.GlanceAPIs[name]`).

Reply:
I just haven't applied glanceAPI back to the instance

Reply:
We only haven't hit that in the tests as they use overrides for network isolation. in the kuttl tests we don't do it